connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java

```diff
@@ -100,32 +103,123 @@ public Uri getAuthorizeUri(Map<String, String> parameters, List<String> locales)
     }
 
     @Override
-    public void onStartAuthorization(final OnStartAuthorizationCallback callback) {
+    public synchronized void onStartAuthorization(
```

Comment:
I really dislike these synchronized methods, including the already present ones. The scope is way to wide, and it's not at all clear what is the actions which needs serialization. Can you rewrite into using explicit locks instead, with as small scope as possible? Alternatively get rid of the volatile usage in favor of explicit state passing between threads, but I think we already discussed that and said it was infeasible in these patches...